### PR TITLE
Add dns-over-quic

### PR DESCRIPTION
**Pull request by wcawijngaards**
This adds a documentation page for dns-over-quic.

### CODE REVIEW

File: source/topics/privacy/dns-over-quic.rst [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+library, but a shared library works too of course.
+
+The ngtcp2 library can be compiled like this:
+
+.. code-block:: bash
+
+    git clone --depth 1 -b v0.19.1 https://github.com/ngtcp2/ngtcp2 ngtcp2
+    cd ngtcp2
+    autoreconf -i
+    ./configure PKG_CONFIG_PATH=$PWD/../openssl+quic_install/lib/pkgconfig LDFLAGS="-Wl,-rpath,$PWD/../openssl+quic_install/lib" --prefix=/path/to/ngtcp2_install
```
Comment (gthess) [inline source/topics/privacy/dns-over-quic.rst:89]: More clear paths for unsuspecting users
```suggestion
    ./configure PKG_CONFIG_PATH=/path/to/openssl+quic_install/lib/pkgconfig LDFLAGS="-Wl,-rpath,/path/to/openssl+quic_install/lib" --prefix=/path/to/ngtcp2_install
```

File: source/topics/privacy/dns-over-quic.rst [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+for DNSSEC.
+
+Compile unbound then like this:
+
+.. code-block:: bash
+
+    ./configure <other flags> --with-ssl=/path/to/openssl+quic_install --with-libngtcp2=/path/to/ngtcp2_install LDFLAGS="-Wl,-rpath -Wl,/path/to/ngtcp2_install/lib" --prefix=/path/to/unbound_install
+    make
+
+Fill in the patch to the openssl+quic install and libngtcp2 install.
```
Comment (gthess) [inline source/topics/privacy/dns-over-quic.rst:109]:
```suggestion
Fill in the path to the openssl+quic install and libngtcp2 install.
```

**Comment by AlexanderBand** [comment]
I enabled PR builds on ReadtheDocs so a preview is [avalable here](https://unbound--50.org.readthedocs.build/en/50/topics/privacy/dns-over-quic.html).

**Review by gthess — APPROVED** [review]
Very clear description and instructions (will also double check later).
Some nits going through it.

File: source/topics/privacy/dns-over-quic.rst (hunk 1 above)
Comment (wcawijngaards) [reply source/topics/privacy/dns-over-quic.rst]: That is nicer, thanks. Applied it.

File: source/topics/privacy/dns-over-quic.rst (hunk 2 above)
Comment (wcawijngaards) [reply source/topics/privacy/dns-over-quic.rst]: Applied the change to fix that mistake. Thank you for the comment.

**Pull request merged**